### PR TITLE
Use FontAwesome for icons

### PR DESCRIPTION
**Pull request by alisan617**
Refs [TNL-4697](https://openedx.atlassian.net/browse/TNL-4697)

Preview: [http://ux-test.edx.org/alisan/fa-icon-tnl-4697/](http://ux-test.edx.org/alisan/fa-icon-tnl-4697/)

Bumps minor version of UXPL.

[UXTK PR is here.](https://github.com/edx/edx-ui-toolkit/pull/73)

## Testing Checklist
- [ ] Write unit tests for all new features.
- [ ] Write accessibility (a11y) tests for all new UI.
- [ ] Manually test responsive behavior.

## Non-testing Checklist
- [x] Consider any documentation your change might need, and which users will be affected by this change.
- [ ] Consult with the Pattern Library team if adding a new component.

## Post-review
- [ ] Squash commits into discrete sets of changes with descriptive commit messages.

## Reviewers
- [ ] @andy-armstrong  
- [ ] @AlasdairSwan 
- [x] @clrux 
- [ ] @cahrens 
- [x] a11y review

If you've been tagged for review, please check your corresponding box once you've given the :+1:.

We'll also automatically suggest reviewers for this PR based on the code it touches.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @alisan617! I've created [OSPR-1286](https://openedx.atlassian.net/browse/OSPR-1286) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: pattern-library/js/edx-icons.js [hunk 1]
```diff
@@ -1,18 +1 @@
...
-            glyphs: '\uE621\uE622\uE623'
-        });
-    });
-}).call(
-    this,
-    typeof define === 'function' && define.amd ? define :
-        (typeof RequireJS !== 'undefined' ? RequireJS.define :
-            edx.GlobalLoader.defineAs('edxicons', 'edx-pattern-library/js/edx-icons.js'))
-);
+require("font-awesome-loader");
```
Comment (AlasdairSwan) [inline pattern-library/js/edx-icons.js]: Nitpick: ```'``` instead of ```"``` for JavaScript strings.

**Comment by AlasdairSwan** [comment]
Just 1 nitpick. Once commits squashed 👍

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "edx-pattern-library",
-  "version": "0.13.0",
+  "version": "0.13.1",
```
Comment (andy-armstrong) [inline package.json]: Note that we update the version number only in a release PR. See https://openedx.atlassian.net/wiki/display/FEDX/UI+Toolkit+release+process

File: pattern-library/js/edx-icons.js [hunk 3]
```diff
@@ -1,18 +1 @@
...
-            glyphs: '\uE621\uE622\uE623'
-        });
-    });
-}).call(
-    this,
-    typeof define === 'function' && define.amd ? define :
-        (typeof RequireJS !== 'undefined' ? RequireJS.define :
-            edx.GlobalLoader.defineAs('edxicons', 'edx-pattern-library/js/edx-icons.js'))
-);
+require('font-awesome-loader');
```
Comment (andy-armstrong) [inline pattern-library/js/edx-icons.js]: I think you can just remove this file, and change the pldoc to have this dependency instead. It doesn't make sense to have this file here because it will be included in the npm package, but clients who use the Pattern Library might not be using Webpack.

File: pattern-library/js/modernizr-custom.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /*!
```
Comment (andy-armstrong) [inline pattern-library/js/modernizr-custom.js]: Do we still need Modernizr? Can we just remove this?

File: pldoc/_design_elements/icons.md [hunk 5]
```diff
@@ -37,148 +37,113 @@ info:               Icons are used to clearly and visually represent some inform
...
-                <span class="text">Warning</span>
-            </span>
-        </div>
-        <div class="use-description">
-            <p>Icon with image fallback</p>
-        </div>
-    </div>
 </div>
 
 <h3 class="hd-6 example-set-hd">Media</h3>
```
Comment (andy-armstrong) [inline pldoc/_design_elements/icons.md]: I'm thinking you should remove all of these examples. This was a demonstration of the edx-icon set, but now we support the full Font Awesome set of icons. This enumeration doesn't really make any sense to me any more.

Comment (andy-armstrong) [inline pldoc/_design_elements/icons.md]: I would also add a paragraph to the opening description saying that we use Font Awesome, and provide a link to its icon set. It would probably also be worth saying something about accessibility and maybe linking to http://fontawesome.io/accessibility/

File: pldoc/_includes/examples/dropdown-menu-js.html [hunk 6]
```diff
@@ -53,7 +53,7 @@
                 url: '/'
             },
             button: {
-                icon: 'icon-angle-down',
+                icon: 'fa-angle-down',
```
Comment (andy-armstrong) [inline pldoc/_includes/examples/dropdown-menu-js.html]: Do you need to also include ``fa`` because the dropdown doesn't add it by default.

**Comment by andy-armstrong** [comment]
@alisan617 This is an awesome first PR! I think you've covered all the changes that I was hoping to see. I have a few small comments so let's discuss how to resolve them.

**Comment by andy-armstrong** [comment]
@alisan617 BTW, I'm unable to generate a preview from your branch. I ran ``npm install`` and then ``gulp preview`` and got this error:

```
$ gulp preview
[19:48:40] Using gulpfile ~/edx/src/ux-pattern-library/gulpfile.js
[19:48:40] Starting 'preview'...
[19:48:40] Starting 'jekyll-build-preview'...
Generating preview for branch alisan/fa-icon-tnl-4697
WARN: Unresolved specs during Gem::Specification.reset:
      jekyll-watch (~> 1.1)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
[19:48:42] Finished 'jekyll-build-preview' after 1.52 s
[19:48:42] Starting 'preview-scripts'...

stream.js:74
      throw er; // Unhandled stream error in pipe.
      ^
Error: ./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js
Module not found: Error: Cannot resolve module 'file-loader' in /Users/andya/edx/src/ux-pattern-library/node_modules/font-awesome-loader
resolve module file-loader in /Users/andya/edx/src/ux-pattern-library/node_modules/font-awesome-loader
  looking for modules in /Users/andya/edx/src/ux-pattern-library/node_modules
    module variation file-loader-webpack-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader doesn't exist (module as directory)
      resolve 'file' file-loader-webpack-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.js doesn't exist
    module variation file-loader-web-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader doesn't exist (module as directory)
      resolve 'file' file-loader-web-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.js doesn't exist
    module variation file-loader-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader doesn't exist (module as directory)
      resolve 'file' file-loader-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.js doesn't exist
    module variation file-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader doesn't exist (module as directory)
      resolve 'file' file-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.js doesn't exist
  looking for modules in /node_modules
    module variation file-loader-webpack-loader
      /node_modules/file-loader-webpack-loader doesn't exist (module as directory)
      resolve 'file' file-loader-webpack-loader in /node_modules
        resolve file
          /node_modules/file-loader-webpack-loader doesn't exist
          /node_modules/file-loader-webpack-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader-webpack-loader.web-loader.js doesn't exist
          /node_modules/file-loader-webpack-loader.loader.js doesn't exist
          /node_modules/file-loader-webpack-loader.js doesn't exist
    module variation file-loader-web-loader
      /node_modules/file-loader-web-loader doesn't exist (module as directory)
      resolve 'file' file-loader-web-loader in /node_modules
        resolve file
          /node_modules/file-loader-web-loader doesn't exist
          /node_modules/file-loader-web-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader-web-loader.web-loader.js doesn't exist
          /node_modules/file-loader-web-loader.loader.js doesn't exist
          /node_modules/file-loader-web-loader.js doesn't exist
    module variation file-loader-loader
      /node_modules/file-loader-loader doesn't exist (module as directory)
      resolve 'file' file-loader-loader in /node_modules
        resolve file
          /node_modules/file-loader-loader doesn't exist
          /node_modules/file-loader-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader-loader.web-loader.js doesn't exist
          /node_modules/file-loader-loader.loader.js doesn't exist
          /node_modules/file-loader-loader.js doesn't exist
    module variation file-loader
      /node_modules/file-loader doesn't exist (module as directory)
      resolve 'file' file-loader in /node_modules
        resolve file
          /node_modules/file-loader doesn't exist
          /node_modules/file-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader.web-loader.js doesn't exist
          /node_modules/file-loader.loader.js doesn't exist
          /node_modules/file-loader.js doesn't exist
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.js]
[/node_modules/file-loader-webpack-loader]
[/node_modules/file-loader-webpack-loader]
[/node_modules/file-loader-webpack-loader.webpack-loader.js]
[/node_modules/file-loader-webpack-loader.web-loader.js]
[/node_modules/file-loader-webpack-loader.loader.js]
[/node_modules/file-loader-webpack-loader.js]
[/node_modules/file-loader-web-loader]
[/node_modules/file-loader-web-loader]
[/node_modules/file-loader-web-loader.webpack-loader.js]
[/node_modules/file-loader-web-loader.web-loader.js]
[/node_modules/file-loader-web-loader.loader.js]
[/node_modules/file-loader-web-loader.js]
[/node_modules/file-loader-loader]
[/node_modules/file-loader-loader]
[/node_modules/file-loader-loader.webpack-loader.js]
[/node_modules/file-loader-loader.web-loader.js]
[/node_modules/file-loader-loader.loader.js]
[/node_modules/file-loader-loader.js]
[/node_modules/file-loader]
[/node_modules/file-loader]
[/node_modules/file-loader.webpack-loader.js]
[/node_modules/file-loader.web-loader.js]
[/node_modules/file-loader.loader.js]
[/node_modules/file-loader.js]
 @ ./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js 6:36824-36886./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js
Module not found: Error: Cannot resolve module 'file-loader' in /Users/andya/edx/src/ux-pattern-library/node_modules/font-awesome-loader
resolve module file-loader in /Users/andya/edx/src/ux-pattern-library/node_modules/font-awesome-loader
  looking for modules in /Users/andya/edx/src/ux-pattern-library/node_modules
    module variation file-loader-webpack-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader doesn't exist (module as directory)
      resolve 'file' file-loader-webpack-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.js doesn't exist
    module variation file-loader-web-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader doesn't exist (module as directory)
      resolve 'file' file-loader-web-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.js doesn't exist
    module variation file-loader-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader doesn't exist (module as directory)
      resolve 'file' file-loader-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.js doesn't exist
    module variation file-loader
      /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader doesn't exist (module as directory)
      resolve 'file' file-loader in /Users/andya/edx/src/ux-pattern-library/node_modules
        resolve file
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.webpack-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.web-loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.loader.js doesn't exist
          /Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.js doesn't exist
  looking for modules in /node_modules
    module variation file-loader-webpack-loader
      /node_modules/file-loader-webpack-loader doesn't exist (module as directory)
      resolve 'file' file-loader-webpack-loader in /node_modules
        resolve file
          /node_modules/file-loader-webpack-loader doesn't exist
          /node_modules/file-loader-webpack-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader-webpack-loader.web-loader.js doesn't exist
          /node_modules/file-loader-webpack-loader.loader.js doesn't exist
          /node_modules/file-loader-webpack-loader.js doesn't exist
    module variation file-loader-web-loader
      /node_modules/file-loader-web-loader doesn't exist (module as directory)
      resolve 'file' file-loader-web-loader in /node_modules
        resolve file
          /node_modules/file-loader-web-loader doesn't exist
          /node_modules/file-loader-web-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader-web-loader.web-loader.js doesn't exist
          /node_modules/file-loader-web-loader.loader.js doesn't exist
          /node_modules/file-loader-web-loader.js doesn't exist
    module variation file-loader-loader
      /node_modules/file-loader-loader doesn't exist (module as directory)
      resolve 'file' file-loader-loader in /node_modules
        resolve file
          /node_modules/file-loader-loader doesn't exist
          /node_modules/file-loader-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader-loader.web-loader.js doesn't exist
          /node_modules/file-loader-loader.loader.js doesn't exist
          /node_modules/file-loader-loader.js doesn't exist
    module variation file-loader
      /node_modules/file-loader doesn't exist (module as directory)
      resolve 'file' file-loader in /node_modules
        resolve file
          /node_modules/file-loader doesn't exist
          /node_modules/file-loader.webpack-loader.js doesn't exist
          /node_modules/file-loader.web-loader.js doesn't exist
          /node_modules/file-loader.loader.js doesn't exist
          /node_modules/file-loader.js doesn't exist
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.webpack-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.web-loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.loader.js]
[/Users/andya/edx/src/ux-pattern-library/node_modules/file-loader.js]
[/node_modules/file-loader-webpack-loader]
[/node_modules/file-loader-webpack-loader]
[/node_modules/file-loader-webpack-loader.webpack-loader.js]
[/node_modules/file-loader-webpack-loader.web-loader.js]
[/node_modules/file-loader-webpack-loader.loader.js]
[/node_modules/file-loader-webpack-loader.js]
[/node_modules/file-loader-web-loader]
[/node_modules/file-loader-web-loader]
[/node_modules/file-loader-web-loader.webpack-loader.js]
[/node_modules/file-loader-web-loader.web-loader.js]
[/node_modules/file-loader-web-loader.loader.js]
[/node_modules/file-loader-web-loader.js]
[/node_modules/file-loader-loader]
[/node_modules/file-loader-loader]
[/node_modules/file-loader-loader.webpack-loader.js]
[/node_modules/file-loader-loader.web-loader.js]
[/node_modules/file-loader-loader.loader.js]
[/node_modules/file-loader-loader.js]
[/node_modules/file-loader]
[/node_modules/file-loader]
[/node_modules/file-loader.webpack-loader.js]
[/node_modules/file-loader.web-loader.js]
[/node_modules/file-loader.loader.js]
[/node_modules/file-loader.js]
 @ ./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js 6:36909-36963./~/font-awesome/fonts/fontawesome-webfont.woff2?v=4.6.3
Module build failed: Error: Cannot find module 'file-loader'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.module.exports (/Users/andya/edx/src/ux-pattern-library/node_modules/url-loader/index.js:18:20)
 @ ./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js 6:37023-37087./~/font-awesome/fonts/fontawesome-webfont.woff?v=4.6.3
Module build failed: Error: Cannot find module 'file-loader'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.module.exports (/Users/andya/edx/src/ux-pattern-library/node_modules/url-loader/index.js:18:20)
 @ ./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js 6:37120-37183./~/font-awesome/fonts/fontawesome-webfont.svg?v=4.6.3
Module build failed: Error: Cannot find module 'file-loader'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.module.exports (/Users/andya/edx/src/ux-pattern-library/node_modules/url-loader/index.js:18:20)
 @ ./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js 6:37313-37375./~/font-awesome/fonts/fontawesome-webfont.ttf?v=4.6.3
Module build failed: Error: Cannot find module 'file-loader'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.module.exports (/Users/andya/edx/src/ux-pattern-library/node_modules/url-loader/index.js:18:20)
 @ ./~/css-loader!./~/sass-loader!./~/font-awesome-loader/font-awesome-styles.loader.js!./~/font-awesome-loader/font-awesome.config.js 6:37215-37277
```

File: AUTHORS [hunk 7]
```diff
@@ -8,3 +8,4 @@ Renzo Lucioni <renzo@edx.org>
 Dennis Jen <djen@edx.org>
 Andy Armstrong <andya@edx.org>
 Alasdair Swan <aswan@edx.org>
+Alisan Tang >atang@edx.org>
```
Comment (cahrens) [inline AUTHORS]: Nit, fix ">" before your email address.

File: pattern-library/js/modernizr-custom.js (hunk 4 above)
Comment (cahrens) [inline pattern-library/js/modernizr-custom.js]: It was my understanding that we can remove Modernizr.

File: pldoc/_design_elements/icons.md (hunk 5 above)
Comment (cahrens) [inline pldoc/_design_elements/icons.md]: I think that @clrux did want to keep some of these examples. Please touch base with him.

**Comment by clrux** [comment]
Let's keep the examples, but let's include a note that there are more options via the Font Awesome library. We're using version 4.2 I think, so let's link to that. Maybe place this link at the top and/or bottom of the list of icons. @andy-armstrong thoughts?

**Comment by alisan617** [comment]
ready for re-review

**Comment by andy-armstrong** [comment]
@clrux So is the idea that this is an edX recommended subset of Font Awesome icons that you should use in preference to the others? I think if we describe that clearly then that works for me.

**Comment by clrux** [comment]
@andy-armstrong Correct. The entire Font Awesome library is huge and I think it'd be good if we listed the items that we use and recommend those icons be what folks choose from for consistency. But the entire library is ultimately available. This sound OK?

File: gulp/tasks/demo_styles.js [hunk 8]
```diff
@@ -10,7 +10,7 @@ gulp.task('demo-styles', function() {
         {
             source: config.documentation.rootDemoSassFile,
             targetDirectory: config.documentation.pldocDest,
-            patternLibraryPath: '/' + config.documentation.pldocDest + '/edx-pattern-library'
```
Comment (clrux) [inline gulp/tasks/demo_styles.js]: Is this change intentional?

File: gulp/tasks/pldoc.js [hunk 9]
```diff
@@ -44,7 +44,7 @@ gulp.task('pldoc-styles', function() {
         {
             source: config.documentation.rootSassFile,
             targetDirectory: config.documentation.pldocDest,
-            patternLibraryPath: '/' + config.documentation.pldocDest + '/edx-pattern-library'
```
Comment (clrux) [inline gulp/tasks/pldoc.js]: Is this change intentional?

File: pattern-library/sass/patterns/_icons.scss [hunk 10]
```diff
@@ -1,486 +0,0 @@
...
-    settings: cog,
-    rotate-left: undo,
-    unsorted: sort,
-    visible: show
-);
-
-// ----------------------------
-// #GLOBAL
-// ----------------------------
-.icon {
```
Comment (clrux) [inline pattern-library/sass/patterns/_icons.scss]: We probably want to keep this part, but it may be able to be moved into another partial.

File: pldoc/_components/tables.md [hunk 11]
```diff
@@ -221,28 +221,22 @@ info:           Tables are used for displaying information in tabular format, wh
       <th scope="col" class="sortable sort-current sort-descending">
         First
         <button class="sort action-sort">
-          <span class="icon-fallback icon-fallback-text">
-            <span class="icon icon-sort" aria-hidden="true"></span>
-            <span class="text">Sort<span class="sr">&nbsp;Ascending</span></span>
-          </span>
+            <span class="icon fa fa-sort" aria-hidden="true"></span>
+            <span class="sr-only">Sort<span class="sr">&nbsp;Ascending</span></span>
```
Comment (clrux) [inline pldoc/_components/tables.md]: We don't need both `sr` and `sr-only`. In fact, I think only `sr-only` works. Just wrap them together like this:

`<span class="sr-only"> Sort ascending</span>`

File: pldoc/_components/tables.md [hunk 12]
```diff
@@ -221,28 +221,22 @@ info:           Tables are used for displaying information in tabular format, wh
...
       </th>
       <th scope="col" class="sortable">
         Lineage
         <button class="sort action-sort">
-          <span class="icon-fallback icon-fallback-text">
-            <span class="icon icon-sort" aria-hidden="true"></span>
-            <span class="text">Sort<span class="sr">&nbsp;Ascending</span></span>
-          </span>
+            <span class="icon fa fa-sort" aria-hidden="true"></span>
+            <span class="sr-only">Sort<span class="sr">&nbsp;Ascending</span></span>
```
Comment (clrux) [inline pldoc/_components/tables.md]: We don't need both `sr` and `sr-only`. In fact, I think only `sr-only` works. Just wrap them together like this:

`<span class="sr-only"> Sort ascending</span>`

File: pldoc/_design_elements/icons.md (hunk 5 above)
Comment (clrux) [inline pldoc/_design_elements/icons.md]: I think it's a good idea to list the icons that we use commonly to aid future consistency with use. What I don't want is three different icons representing "edit" or something like that.

File: gulp/tasks/demo_styles.js (hunk 8 above)
Comment (alisan617) [inline gulp/tasks/demo_styles.js]: Yes. Please see pr #333 for details.

File: gulp/tasks/pldoc.js (hunk 9 above)
Comment (alisan617) [inline gulp/tasks/pldoc.js]: Yes. Please see pr #333 for details.

File: pattern-library/sass/patterns/_icons.scss (hunk 10 above)
Comment (alisan617) [inline pattern-library/sass/patterns/_icons.scss]: added icons.scss back but only with the icon class

File: pattern-library/sass/_edx-pattern-library.scss [hunk 13]
```diff
@@ -18,9 +18,9 @@
 @import 'patterns/lists';
 @import 'patterns/forms';
 @import 'patterns/grid';
+@import 'patterns/icons';
```
Comment (cahrens) [inline pattern-library/sass/_edx-pattern-library.scss]: @alisan617 Should this still be here?

**Comment by cahrens** [comment]
@alisan617 This looks good from my perspective-- great work! 👍 

Please do get thumbs-ups from @andy-armstrong and @clrux before merging, and remember to squash your commits before you merge.

**Comment by clrux** [comment]
👍

**Comment by andy-armstrong** [comment]
@alisan617 I've pulled your latest changes and still can't generate a preview:

```
Module build failed: Error: Cannot find module 'file-loader'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.module.exports (/Users/andya/edx/src/ux-pattern-library/node_modules/url-loader/index.js:18:20)
```

Any ideas? @AlasdairSwan, does it work for you?

👍 once we can work out what this issue is.

**Comment by alisan617** [comment]
@andy-armstrong My bad. I added the file loader back to package json. Once you npm install, it should be good.

**Comment by cahrens** [comment]
Note that this PR removes edx-icons, modernizer, etc. entirely. This means we can't upgraded the version of the pattern library used by edx-platform until edx-icons are entirely removed.

@alisan617 will be taking on this work in the next sprint. Are there any concerns from @andy-armstrong or @AlasdairSwan about merging this ahead of the necessary edx-platform work?

**Comment by andy-armstrong** [comment]
@alisan617 Thanks, the preview works now. 👍 from me.

@cahrens I agree that there's a slight risk if this merges that we won't be able to fix issues for the platform as we won't be able to upgrade. Can you close the story since the work is complete, but just not push the merge button until we're ready to consume it in edx-platform?

@dsjen does this affect Inisghts at all? I don't think you're using it yet, but just wanted to check.

**Comment by dsjen** [comment]
Thanks for checking!  We're not using icons through the PL, but are indeed using FA icons.

**Comment by cahrens** [comment]
@andy-armstrong That sounds like a good plan. @alisan617 please squash the commits on this branch into a single commit, and then leave the PR open for now. I'll mark the story as done.

**Comment by alisan617** [comment]
@cahrens Commits squashed.

**Comment by bjacobel** [comment]
This is a blocker for #334, do we have a rough timeline for when this will merge?

**Comment by alisan617** [comment]
@bjacobel I am about to finish https://openedx.atlassian.net/browse/TNL-4813 story, so hopefully by tomorrow.

**Comment by cahrens** [comment]
👍 On temporarily putting the library files back in. After edx-platforms and programs are completely edx-icons free, then the libraries can be removed for good.

**Comment by andy-armstrong** [comment]
Ugggh, it is a shame to have to add back all the icons too. 👍  while holding my nose. :-)
